Data/IntSet/Internal.hs

```diff
@@ -562,66 +567,68 @@ difference Nil _     = Nil
 -- | /O(n+m)/. The intersection of two sets.
 intersection :: IntSet -> IntSet -> IntSet
 intersection t1@(Bin p1 m1 l1 r1) t2@(Bin p2 m2 l2 r2)
-  | shorter m1 m2  = intersection1
-  | shorter m2 m1  = intersection2
-  | p1 == p2       = bin p1 m1 (intersection l1 l2) (intersection r1 r2)
-  | otherwise      = Nil
+  | shorter m1 m2 = intersection1
+  | shorter m2 m1 = intersection2
+  | p1 == p2      = bin p1 m1 (intersection l1 l2) (intersection r1 r2)
+  | otherwise     = Nil
   where
-    intersection1 | nomatch p2 p1 m1  = Nil
-                  | zero p2 m1        = intersection l1 t2
-                  | otherwise         = intersection r1 t2
+    intersection1 | nomatch p2 p1 m1 = Nil
+                  | zero p2 m1       = intersection l1 t2
+                  | otherwise        = intersection r1 t2
 
-    intersection2 | nomatch p1 p2 m2  = Nil
-                  | zero p1 m2        = intersection t1 l2
-                  | otherwise         = intersection t1 r2
+    intersection2 | nomatch p1 p2 m2 = Nil
+                  | zero p1 m2       = intersection t1 l2
+                  | otherwise        = intersection t1 r2
 
 intersection t1@(Bin _ _ _ _) (Tip kx2 bm2) = intersectBM t1
-  where intersectBM (Bin p1 m1 l1 r1) | nomatch kx2 p1 m1 = Nil
-                                      | zero kx2 m1       = intersectBM l1
-                                      | otherwise         = intersectBM r1
-        intersectBM (Tip kx1 bm1) | kx1 == kx2 = tip kx1 (bm1 .&. bm2)
-                                  | otherwise = Nil
-        intersectBM Nil = Nil
+  where
+    intersectBM (Bin p1 m1 l1 r1) | nomatch kx2 p1 m1 = Nil
+                                  | zero kx2 m1       = intersectBM l1
+                                  | otherwise         = intersectBM r1
+    intersectBM (Tip kx1 bm1) | kx1 == kx2 = tip kx1 (bm1 .&. bm2)
+                              | otherwise  = Nil
+    intersectBM Nil = Nil
 
 intersection (Bin _ _ _ _) Nil = Nil
 
 intersection (Tip kx1 bm1) t2 = intersectBM t2
-  where intersectBM (Bin p2 m2 l2 r2) | nomatch kx1 p2 m2 = Nil
-                                      | zero kx1 m2       = intersectBM l2
-                                      | otherwise         = intersectBM r2
-        intersectBM (Tip kx2 bm2) | kx1 == kx2 = tip kx1 (bm1 .&. bm2)
-                                  | otherwise = Nil
-        intersectBM Nil = Nil
+  where
+    intersectBM (Bin p2 m2 l2 r2) | nomatch kx1 p2 m2 = Nil
```

Comment:
Do we want to move the guards to the next line?

Reply:
I personally would prefer to always hang them, I wasn't able to find a configuration option to force that though. I think the algorithm only hangs them if one of them would overflow the 80 column limit. I'll look a bit more and submit an issue if I can't find one.